lib/ansible/modules/windows/win_user_right.ps1

```diff
@@ -0,0 +1,354 @@
+    $new_users = Build-NewUserList -action $action -users $users -existing_users $existing_users
+
+    # sort the user objects for later comparison and remove duplicates
+    $new_users = $new_users | Sort-Object -Unique
+    $existing_users = $existing_users | Sort-Object -Unique
+
+    # compare the list and make changes as necessary
+    if ($new_users.Length -eq 0) {
+        $will_change = $true
+        $secedit_ini.$ini_right_key.$name = $null
+    }
+
+    $comparison = Compare-List -existing_list $existing_users -new_list $new_users
+    if ($comparison.mismatch -eq $true) {
+        foreach ($added in $comparison.added) {
+            $result.added += $added
+        }
+        foreach ($removed in $comparison.removed) {
+            $result.removed += $removed
+        }
+        $will_change = $true
+        $secedit_ini.$ini_right_key.$name = $new_users -join ","
+    }
+} else {
+    if ($users.Length -gt 0 -and (@("add", "set") -contains $action)) {
+        $will_change = $true
+        $new_users = @()
+        foreach ($user in $users) {
+            $username = Get-Username -sid $user
+            $result.added += $username
+            $new_users += "*$(Get-SID -account_name $user)"
+        }
+        $secedit_ini.$ini_right_key.$name = $new_users -join ","
+    }
+}
+
+if ($will_change) {
+    $ini_contents = ConvertTo-Ini -ini $secedit_ini
+    Set-Content -Path $secedit_ini_path -Value $ini_contents -Encoding Unicode -WhatIf:$check_mode
+    $result.changed = $true
+
+    if (-not $check_mode) {
+        $secedit_db_path = [IO.Path]::GetTempFileName()
+        Remove-Item -Path $secedit_db_path -Force # needs to be deleted for SecEdit.exe /import to work
+
+        $import_result = Run-SecEdit -arguments @("/configure", "/db", $secedit_db_path, "/cfg", $secedit_ini_path, "/quiet")
+        $result.changed = $true
+        $result.import_log = $import_result.log
+        Remove-Item -Path $secedit_ini_path -Force
+        if ($import_result.rc -ne 0) {
+            Fail-Json $result "Failed to import secedit.ini file from $($secedit_ini_path).`nRC: $($import_result.rc)`nSTDOUT: $($import_result.stdout)`nSTDERR: $($import_result.stderr)`nLOG: $($import_result.log)"
+        }
+
+        # secedit doesn't error out on improper entries, re-export and verify
+        # the changes occurred
+        $export_result = Run-SecEdit -arguments @("/export", "/cfg", $secedit_ini_path, "/quiet")
+
+        # check the return code and if the file has been populated, otherwise error out
+        if (($export_result.rc -ne 0) -or ((Get-Item -Path $secedit_ini_path).Length -eq 0)) {
+            Remove-Item -Path $secedit_ini_path # file is empty and we don't need it
+            Fail-Json $result "Failed to export secedit.ini file to $($secedit_ini_path).`nRC: $($export_result.rc)`nSTDOUT: $($export_result.stdout)`nSTDERR: $($export_result.stderr)`nLOG: $($export_result.log)"
```

Comment:
Do we want to include all that in the message, or just return discrete keys in the fail dict? Could see reasons for either...